src/main/java/com/endava/endabank/security/UserAuthentication.java

```diff
@@ -0,0 +1,56 @@
+package com.endava.endabank.security;
+
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+
+import java.util.Collection;
+import java.util.Objects;
+
+
+public class UserAuthentication extends User {
+    private Integer id;
+    private Boolean isApproved;
+    private Boolean isEmailVerified;
+
+    public UserAuthentication(String username, String password,
+                              Collection<? extends GrantedAuthority> authorities,
+                              Integer id, Boolean isApproved, Boolean isEmailVerified) {
+        super(username, password, authorities);
+        this.id = id;
+        this.isApproved = isApproved != null && isApproved;
+        this.isEmailVerified = isEmailVerified;
+    }
+
+    public Integer getId() {
```

Comment:
Hi,

Please check if it is possible to replace these "get" methods for @data @get @set